Content.Server/Fluids/EntitySystems/MoppingSystem.cs

```diff
@@ -0,0 +1,350 @@
+                && currentVolume <= 0 ) // tool is dry
+        {
+            // These return conditions will abort BEFORE the do_after is called:
+            if (!_solutionSystem.TryGetDrainableSolution(target, out var drainableSolution))
+            {
+                return;
+            }
+            else if (drainableSolution.CurrentVolume <= 0) // target container is empty (liquid source)
+            {
+                msg = "mopping-system-target-container-empty";
+                user.PopupMessage(user, Loc.GetString(msg, ("target", target))); // play message now because we are returning.
+                return;
+            }
+            else
+            {
+                // Determine transferAmount
+                transferAmount = FixedPoint2.Min(availableVolume * 0.5, drainableSolution.CurrentVolume); // Let's transfer up to to half the tool's available capacity to the tool.
+
+                donor = target; // the drainable container's Uid
+                donorSolutionName = drainable.Solution;
+
+                acceptor  = used; // the mop/tool Uid
+                acceptorSolutionName = "absorbed"; // by definition on AbsorbentComponent
+
+                // Set delay/popup/sound if nondefault. Popup and sound will only play on a successful doAfter.
+                // default delay is fine for this case.
+                msg = "mopping-system-drainable-success";
+                sfx = component.TransferSound;
+            }
+        }
+        else return;
+
+        var doAfterArgs = new DoAfterEventArgs(user, delay, target: target)
+        {
+            BreakOnUserMove = true,
+            BreakOnStun = true,
+            BreakOnDamage = true,
+            MovementThreshold = 0.2f,
+            BroadcastCancelledEvent = new TransferCancelledEvent()
+            {
+                Target = target,
+                Component = component // (the AbsorbentComponent)
+            },
+            BroadcastFinishedEvent = new TransferCompleteEvent()
+            {
+                User = user,
+                Tool = used,
+                Target = target,
+                Donor = donor,
+                Acceptor = acceptor,
+                Component = component,
+                DonorSolutionName = donorSolutionName,
+                AcceptorSolutionName = acceptorSolutionName,
+                Message = msg,
+                Sound = sfx,
+                TransferAmount = transferAmount
+            }
+        };
+
+        // Can't interact with too many entities at once.
+        if (component.MaxInteractingEntities < component.InteractingEntities.Count + 1)
+            return;
+
+        // Can't interact with the same container multiple times at once.
+        if (!component.InteractingEntities.Add(target))
+            return;
+
+        var result = _doAfterSystem.WaitDoAfter(doAfterArgs);
+    }
+
+
+    private void OnTransferComplete(TransferCompleteEvent ev)
```

Comment:
Bonus points if you can make these events static.

Reply:
Thanks - what would that do? I'm not sure when to use static in general.

Making these three static gives me the error below. This is new territory for me - any help you can give is appreciated!
![image](https://user-images.githubusercontent.com/97707302/156909683-27435ecf-d846-4644-8805-c2560fb18314.png)